react_juce/core/ImageView.cpp

```diff
@@ -34,47 +34,58 @@ namespace reactjuce
         if (name == sourceProp)
         {
             const juce::String source = value.toString();
-            const juce::URL    sourceURL = source;
+            const int sourceHash      = juce::DefaultHashFunctions::generateHash(source, 1024);
+            const juce::URL sourceURL = source;
 
-            if (isWellFormedURL(sourceURL))
+            // No need to recalculate the image if it's already in the ImageCache.
+            const juce::Image& cachedImage = juce::ImageCache::getFromHashCode(sourceHash);
+            if (cachedImage.isValid())
             {
-                // Web images are downloaded in a separate thread to avoid blocking.
-                if (sourceURL.isProbablyAWebsiteURL(sourceURL.toString(false)))
+                setDrawableImage(cachedImage, sourceHash);
+            }
+            else
+            {
+                try
                 {
-                    const auto sourceHash = juce::DefaultHashFunctions::generateHash(source, 1024);
-                    const auto& cachedImage = juce::ImageCache::getFromHashCode(sourceHash);
-                    if (cachedImage.isValid())
+                    if (isWellFormedURL(sourceURL))
```

Comment:
Same here I think we can collapse some of this with early returns.

```
if (sourceURL.isLocalFile())
    return setDrawableImage(loadImageFromFileURL(sourceURL));
    
if (sourceURL.isProbablyAWebsiteURL(source) && isWellFormedURL(sourceURL))
    return downloadImageAsync(source);

if (sourceURL.startsWith("data:image/")
    return setDrawableImage(....);
    
throw std::logic_error("Unsupported image URL:" + source);
```

That feels a bit neater to me. What do you think?